lib/cfn-model/parser/cfn_parser.rb

```diff
@@ -31,7 +31,13 @@ class CfnParser
   # Given raw json/yml CloudFormation template, returns a CfnModel object
   # or raise ParserErrors if something is amiss with the format
   def parse(cloudformation_yml, parameter_values_json=nil)
-    cfn_hash = pre_validate_model cloudformation_yml
```

Comment:
the validator finds badness and then returns the cfn_hash.  what is it the TransformerRegistry needs to access before the yml/json has been "pre-validated"?

Reply:
It needs to create / modify resources in the YAML hash before higher level validation (like references being defined) can be successfully performed